### DISCUSSION TITLE
NATS in the cluster deletes messages in the 'nak timeout' state when updating the stream.

### DISCUSSION THREAD

**Discussion by AlexandrTQ** [nats-io/nats-server#5511] [Q&A]
When we used one nats instance, there was no such problem.
Creating/updating a stream:
```
	stream, err := js.CreateOrUpdateStream(ctx, jetstream.StreamConfig{
		Name:       c.StreamName,
		Retention:  jetstream.WorkQueuePolicy,
		Subjects:   []string("example.>"),
		Storage:    jetstream.FileStorage,
		Duplicates: time.Minute * 2,
		Replicas:   3,
	})
```
Creating/updating consumer:
```
	cons, err := nw.stream.CreateOrUpdateConsumer(ctx, jetstream.ConsumerConfig{
		Durable:       consumerName,
		FilterSubject: "example.default",
		MaxDeliver:    opt.MaxDeliver,
		MaxAckPending: opt.MaxAckPending,
		AckWait:       time.Duration(opt.AckWaitSeconds) * time.Second,
		AckPolicy:     jetstream.AckExplicitPolicy,
		DeliverPolicy: jetstream.DeliverAllPolicy,
	})

	cons.Consume(func(msg jetstream.Msg) { msg.NakWithDelay(time.Hour) })
```

The general flow is:
1) Service A created a stream and a consumer.
2) Service B updated the stream using the same settings and began publishing in subject.
3) Service A delayed some messages using NakWithDelay
4) Service B restarted and executed CreateOrUpdateStream again with the same settings.
5) All pending messages have been deleted.
Here is the metric nats_consumer_num_ack_pending
![image](https://github.com/nats-io/nats-server/assets/90096536/b89c9f54-c148-4f2b-af9d-8889477eec53)

Please tell me why this happens.

**Comment by derekcollison** [comment]
When you say one nats instance, you mean one server?
